filewatch/filters.go

```diff
@@ -824,8 +826,31 @@ func initStateFile(p string) (fout *os.File, states map[FileName]*int64, err err
 	}
 	if fi.Size() > 0 {
 		if err = gob.NewDecoder(fout).Decode(&states); err != nil {
-			err = fmt.Errorf("Failed to load existing states: %v", err)
-			return
+			// hold onto the decode error in case we can't get to a backup
+			serr := err
+			fout.Close()
+
+			// find a suitable backup filename
+			fname := p + backupSuffix
+			if _, err := os.Stat(fname); err != nil {
+				// we have to start counting
+				var count int
+				for {
+					fname = fmt.Sprintf("%v%v%d", p, backupSuffix, count)
+					if _, err := os.Stat(fname); os.IsNotExist(err) {
```

Comment:
set a maximum here.  If by some crazy situation you can stat the first name but not the one with the number (symlinks... or whatever... then this will never stop.

Rather than attaching a number, I would recommend a backup file that appends a timestamp, if that fails... quit.

Reply:
numbers are easier to grok, and the files have timestamps I'd argue. Limits definitely.

Reply:
sounds good to me.